test/test_cuda.py

```diff
@@ -2133,15 +2133,19 @@ def _compare_trilu_indices(self, row, col, offset=0, dtype=torch.long):
         else:
             # explicitly convert 'cpu' tensor to 'cuda' to avoid a bug in tril
             # and triu cuda kernel (see #15226)
-            self.assertEqual(
-                torch.ones(row, col, dtype=dtype, device='cpu')
-                     .tril(offset).nonzero().transpose(0, 1).cuda(),
-                torch.tril_indices(row, col, offset, dtype=dtype, device='cuda'))
+            x = torch.ones(row, col, dtype=dtype, device='cpu') \
+                     .tril(offset).nonzero().transpose(0, 1).cuda()
+            torch.cuda.synchronize()
```

Comment:
You don't need to synchronize here. self.assertEqual should have a sync point somewhere